### DISCUSSION TITLE
Connection status awareness

### DISCUSSION THREAD

**Discussion by mfodor** [prisma/prisma#13978] [Q&A] [answered]
Is there a way to be aware of connection status with Prisma?

I understand it does everything automatically so I don't have to worry about anything related to connection.  
But what if I want to?

Basically I miss two things:

- Catchable event. Either `client.on('disconnected', ...)` or implementing an interface (e.g. `onDisconnected() { ... }`)
- `$connect()` throwing error if it can not connect. No exception raised when DB is not started and I start the application.

```typescript
// Context:
//   - DB not started yet

try {
  await client.$connect();
  console.log('DB connected');
} catch (e) {
  console.log('DB is unavailable');
}

// Output
//
// > DB connected
```

My use case: I would like to send a system message to maintainers and shut the whole service down if the DB stopped and could not recover connection within a time frame.

I tried to find the answer here and over the internet but could not find it.

Related official docs:
- https://www.prisma.io/docs/reference/api-reference/prisma-client-reference#connect-1
- https://www.prisma.io/docs/concepts/components/prisma-client/working-with-prismaclient/connection-management
- https://www.prisma.io/docs/guides/performance-and-optimization/connection-management

SO question I opened:
- https://stackoverflow.com/questions/72730216/connection-awareness

**Comment by mfodor** [comment, marked as answer]
Okay, I was able to overcome it. I guess it's rather a bug than a feature.

So `$connect()` does not throw error (does not reject) when it could already connect successfully before but the DB has been stopped meanwhile and `$disconnect()` was not called.

So, calling `$disconnect()` before calling `$connect()` when lost connection is recognized resulted in `$connect()` throwing error (rejecting) again.

I still miss the ability to watch for connection events but this is enough for me now.

**Reply by austincrim** [reply]
Hey @mfodor!

Glad you can workaround this for now. Would you mind opening a [feature request](https://github.com/prisma/prisma/issues/new?assignees=&labels=&template=feature_request.md&title=) for how we could better solve this in the future?

**Reply by mfodor** [reply]
Okay, I just opened [this issue](https://github.com/prisma/prisma/issues/13993)

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
